src/ucp/core/ucp_ep.c

```diff
@@ -2812,6 +2812,18 @@ int ucp_ep_do_keepalive(ucp_ep_h ep)
     return 0;
 }
 
+static void ucp_ep_req_purge_send(ucp_request_t *req, ucs_status_t status)
+{
+    if ((status == UCS_OK) ||
+        (ucp_ep_config(req->send.ep)->key.err_mode ==
+         UCP_ERR_HANDLING_MODE_NONE) ||
+        !(req->flags & UCP_REQUEST_FLAG_RKEY_INUSE)) {
+        ucp_request_complete_and_dereg_send(req, status);
+    } else {
+        ucp_request_dt_invalidate(req, status);
```

Comment:
can we invalidate rkey which is in use by other request/ep at this time? for example, if pattern is naive broadcast

Reply:
invalidation here is excluding region from rcache page table, it doesn't unregister region immediately, but waits for all other references are completed